### ISSUE TITLE
GlobalMeterProvider is not returning the MeterProvider created by the agent

### ISSUE THREAD

**Issue by jeiroa** [open-telemetry/opentelemetry-java-instrumentation#3878]
**Describe the bug**
**GlobalMeterProvider.get()** is returning a **NoopMeterProvider** when I am expecting the **SdkMeterProvider** registered by the Java agent.
I am trying this because I would like to add my own ones along with the ones provided by the agent.

**Steps to reproduce**
I am launching a Spring Boot application with the following VM arguments in order to run the Open Telemetry Java agent configured to export Prometheus metrics on localhost, port 9464 and not to export traces:
`-javaagent:extras/opentelemetry-javaagent-all.jar -Dotel.metrics.exporter=prometheus -Dotel.exporter.prometheus.port=9464 -Dotel.traces.exporter=none -Dotel.resource.attributes=service.name=example`

Then, there is a Spring Boot Configuration class that returns the _MetricProvider_ as a Spring bean so it could be injected in any other bean and I will be able to add meters:

```
import io.opentelemetry.api.metrics.GlobalMeterProvider;
import io.opentelemetry.api.metrics.MeterProvider;

@Configuration
public class MetricsConfig {
	@Bean
	public MeterProvider meterProvider() {
		MeterProvider meterProvider = GlobalMeterProvider.get();

		return meterProvider;
	}
}
```

When I start the application I can see the agent is loaded correctly at the very beginning:
`[otel.javaagent 2021-08-19 18:02:11:074 +0200] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.4.1`

As I placed a breakpoint in the return of the code above, I am able to debug the content of _meterProvider_ variable and I can see it is a _NoopMeterProvider_  when I was expecting a _SdkMeterProvider_.

I also set a breakpoint in _GlobalMeterProvider#set_ method to check if the _MeterProvider_ is set during the start-up. It never gets triggered.

**What did you expect to see?**
The _SdkMeterProvider_ created by the agent in order to add more metrics to be available in the Prometheus HTTPServer started by the Prometheus metric exporter configured in the agent on port 9464

**What did you see instead?**
A _NoopMeterProvider_ so custom meters I am creating are not shown in _http://localhost:9464/metrics_.

**What version are you using?**
opentelemetry-javaagent v1.4.1

-- code dependencies --
opentelemetry-api-metrics:1.5.0-alpha
opentelemetry-sdk-metrics:1.5.0-alpha
opentelemetry-bom:1.5.0
opentelemetry-api (version 1.5.0 provided by opentelemetry-bom)
opentelemetry-extension-annotations (version 1.5.0 provided by opentelemetry-bom)
opentelemetry-sdk (version 1.5.0 provided by opentelemetry-bom)

**Environment**
Compiler: AdoptOpenJDK-11.0.11+9
OS: Ubuntu 20.04

**Additional context**
Maybe what I am trying is not the expected behaviour. I am just trying the following recommendation: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3353#issuecomment-864039153
Any help will be much appreciated.

**Comment by jkwatson** [comment]
I see you're trying to combine javaagent v1.4.1 with api/sdk/metrics 1.5.0-alpha. Those versions are definitely very incompatible, as the 1.5.0-alpha metrics API has radically changed. I don't know if that's the root of your issue, but it's definitely something that should not work in any case.

**Comment by jeiroa** [comment]
Hi @jkwatson. You are totally right. I downgraded _opentelemetry-api-metrics_ and _opentelemetry-sdk-metrics_ dependencies to v1.4.1-alpha and the MeterProvider is now the expected one.
I checked that _opentelemetry-bom_ can be on either 1.4.1 or 1.5.0 but it seems that _opentelemetry-*-metrics_ dependencies must be in sync with the agent version.
I am looking forward to downloading version 1.5.0 of the agent.
Thanks for your help.
